### PR TITLE
Fixed error importing survey from collect (SamplingPointDataImportJob)

### PR DESCRIPTION
**Pull request by SteRiccio**
## Types of changes

- [ ] Docs change
- [ ] Refactoring
- [ ] Dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Brief description of the changes proposed and/or how the issue(s) has(have) been solved.

I've modified the way the category import summary is generated for sampling point data by passing 2 parameters to the CategoryImportSummaryGenerator function:
- codeColumnPattern: regular expression used to identify the columns of type code
- ignoreLabelsAndDescriptions: if true, all columns with type other than code will be considered as extra columns

## How has this been tested

Unit test for CategoryImportSummaryGenerator has been created.

##### Do you consider this PR needs further testing?

- [ ] No
- [x] Yes

Several combinations of columns can be added to the test.

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-gk5anb Error importing survey from Collect (sampling point data)](https://app.clickup.com/t/gk5anb)

File: core/geo/point.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const newPoint = ({ srs, x, y }) => ({
  * @returns {object} - The parsed Point object.
  */
 export const parse = (pointText) => {
-  const match = /SRID=((EPSG:)?(\w+));POINT\((\d+(\.\d+)?) (\d+(\.\d+)?)\)/.exec(pointText)
+  const match = /SRID=((EPSG:)?(\w+));POINT\((-?\d+(\.\d+)?) (-?\d+(\.\d+)?)\)/.exec(pointText)
```
Comment (minotogna) [inline core/geo/point.js:25]: should we use arena-core?

File: server/modules/category/categoryValidator.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ const _extraPropValidators = {
     Validator.validateNumber(Validation.messageKeys.categoryEdit.itemExtraPropInvalidNumber, { key })(key, extra),
   [Category.itemExtraDefDataTypes.geometryPoint]: (key, extra) => {
     const point = Point.parse(extra[key])
-    return GeoUtils.isCoordinateValid(Point.getSrs(point), Point.getX(point), Point.getY(point))
+    return point && GeoUtils.isCoordinateValid(Point.getSrs(point), Point.getX(point), Point.getY(point))
       ? null
       : ValidationResult.newInstance(Validation.messageKeys.categoryEdit.itemExtraPropInvalidGeometryPoint, { key })
   },
```
Comment (sorja) [inline server/modules/category/categoryValidator.js:80]: should we use for readability in this kind of cases
```
if (condition) return null
return ValidationResult.newInstance(...)
```

**Review by sorja — COMMENTED** [review]
👍

File: core/geo/point.js (hunk 1 above)
Comment (SteRiccio) [reply core/geo/point.js:25]: ok

File: server/modules/category/categoryValidator.js (hunk 2 above)
Comment (SteRiccio) [reply server/modules/category/categoryValidator.js:80]: ok, in any case this part of the code will be changed with the use of arena-core modules.

File: server/modules/collectImport/service/collectImport/metaImportJobs/samplingPointDataImportJob.js [hunk 3]
```diff
@@ -76,11 +89,14 @@ export default class SamplingPointDataImportJob extends CategoryImportJob {
   extractItemExtraProps(extra) {
     const { srs_id: srs, x, y } = extra
 
-    const point = Point.newPoint({ srs, x, y })
+    const point = PointFactory.createInstance({ srs, x, y })
+
+    const toString = (p) => `SRID=${p.srs};POINT(${p.x} ${p.y})`
```
Comment (minotogna) [inline server/modules/collectImport/service/collectImport/metaImportJobs/samplingPointDataImportJob.js:94]: don't we have Points.toString ?

**Review by minotogna — COMMENTED** [review]
remove @esri dependency

File: server/modules/collectImport/service/collectImport/metaImportJobs/samplingPointDataImportJob.js (hunk 3 above)
Comment (SteRiccio) [reply server/modules/collectImport/service/collectImport/metaImportJobs/samplingPointDataImportJob.js:94]: yes, right

**Comment by SteRiccio** [comment]
> remove @Esri dependency

ok, I'll do it

File: .github/workflows/test.js.yml [hunk 4]
```diff
@@ -53,3 +53,9 @@ jobs:
         run: while ! nc -z localhost 9090; do sleep 5; done;
       - name: Run tests
         run: yarn test
+        env:
+          PGHOST: localhost
+          PGPORT: 5444
+          PGDATABASE: of-arena-test
+          PGUSER: arena
+          PGPASSWORD: arena
```
Comment (minotogna) [inline .github/workflows/test.js.yml]: why is this needed?

File: package.json [hunk 5]
```diff
@@ -96,9 +96,9 @@
     "webpack-node-externals": "^2.5.2"
   },
   "dependencies": {
-    "@esri/proj-codes": "^2.7.0",
     "@godaddy/terminus": "^4.6.0",
-    "@openforis/arena-server": "^0.0.16",
+    "@openforis/arena-core": "^0.0.51",
```
Comment (minotogna) [inline package.json:100]: is this needed?

Comment (SteRiccio) [reply package.json:100]: yes, otherwise you cannot import modules from arena-core

**Review by minotogna — APPROVED** [review]
